repository_service_tuf_worker/repository.py

```diff
@@ -1057,9 +1064,59 @@ def metadata_rotation(
         result = ResultDetails(
             status="Task finished.",
             details={
-                "message": "metadata rotation finished",
+                "message": "metadata update finished",
             },
             last_update=datetime.now(),
         )
 
         return asdict(result)
+
+    def metadata_update(
+        self,
+        payload: Dict[Literal["metadata"], Dict[Literal[Root.type], Any]],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        """
+        Update TUF metadata.
+
+        Args:
+            payload: contains new metadata
+                Supported metadata types: Root
+                example: {"metadata": {"root": Any}}
+            update_state: not used, but required argument by `app.py`
+        """
+
+        # there is also a verification in the RSTUF API calls
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+        if bootstrap is None or "pre-" in bootstrap:
+            raise RepositoryError(
+                "Metadata Update requires a complete bootstrap"
+            )
+
+        metadata = payload.get("metadata")
+        if metadata is None:
+            raise KeyError("No 'metadata' in the payload")
+
+        if Root.type in metadata:
+            new_root = Metadata.from_dict(metadata[Root.type])
+            return self._root_metadata_update(new_root)
+        else:
+            raise ValueError("Unsupported Metadata type")
+
+    def metadata_rotation(
+        self,
+        payload: Dict[Literal["metadata"], Dict[Literal[Root.type], Any]],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> Dict[str, Any]:
+        deprecation_message = (
+            "Function `metadata_rotation` will be deprecated on version "
```

Comment:
nit: IIUC you're deprecating it now and removing it in 1.0.0?

Reply:
Yes. I will fix the sentence.